proofs/proof_global.mli

```diff
@@ -12,16 +12,20 @@
      toplevel. In particular it defines the global proof
      environment. *)
 
+type pstate
 type t
-val get_current_proof_name : t -> Names.Id.t
-val get_current_persistence : t -> Decl_kinds.goal_kind
+
```

Comment:
IMHO you should avoid the `t -> pstate` renaming and instead introduce a `Stack` module so:

- `Proof_global.t` => type of proofs that can be closed
- `Proof_global.Stack.t` => Stack of such proofs [non-empty?]

This can be done later when we separate the lemma-specific stuff from the closable-specific stuff, but likely having this already here will introduce less churn.